### PR TITLE
changed the nesting of figcaption and links to meet a11y guidelines

### LINKED ISSUE #4642: [pni] homepage: un-nest figcaption
The `<figcaption>` that we use in `<figure>` elements for each product cannot be nested inside another element and needs to be a direct child of `<figure>`.

### PR DESCRIPTION
**Pull request by Pomax**
Closes #4642 

Open https://foundation.mozilla.org/en/privacynotincluded/ in the browser and run the AXE accessibility tests: it should give a few tens of `figure` warnings that figures should have a label. Rerun that test on this PR's PNI homepage: there should be no errors for `figure`.

### CODE REVIEW

File: network-api/networkapi/buyersguide/templates/buyersguide_home.html [hunk 1]
```diff
@@ -57,34 +57,41 @@ <h1 class="h1-heading text-center mt-4">{% trans "Be Smart. Shop Safe." %}</h1>
...
+            <img
+              class="product-thumbnail"
+              src="{{mediaUrl}}{{"AWS_LOCATION"|env}}/{{product.image }}"
+              alt="{% blocktrans with product_name=product.name %}link to {{product_name}}{% endblocktrans %}"
+            >
             {% endif %}
-            <figcaption class="d-none d-md-block mt-md-2 text-left">
-            <div class="body-small">{{product.company}}</div>
-            <div class="body">{{product.name}}</div>
-            </figcaption>
```
Comment (mmmavis) [inline network-api/networkapi/buyersguide/templates/buyersguide_home.html:86]: nice catch!

**Comment by Pomax** [comment]
@mmmavis yes, but those are a _different_ error. This PR only solves the figcaption issue, not the fact that there are missing landmarks (header/nav/main/footer).

**Comment by mmmavis** [comment]
Ah kk I probably misunderstood your PR description about no errors for `figure` should be found.

**Comment by mmmavis** [comment]
> Open https://foundation.mozilla.org/en/privacynotincluded/ in the browser and run the AXE accessibility tests: it should give a few tens of figure warnings that figures should have a label. 

@Pomax I just tested the production page. AXE didn't show me any warnings or errors that are related to missing figure labels. All the `figure` related issues I got were landmark related.. Do you have a screencap of the AXE warnings you are getting?

**Comment by Pomax** [comment]
yeah looks like my bad - these errors show up during `cypress a11y`, not in the axe extension in the browser =(

**Comment by mmmavis** [comment]
@Pomax your code changes make sense to me. But I'm having issues spotting the warning/errors from `npm run cypress:a11y`.

1. I checked out to `master` branch to see the "before" scenario
2. Ran `npm run cypress:a11y`
3. Didn't see any `figure` related flags. Was I looking at the wrong place?

Accessibility Tests -- PNI Page Assessments -- macbook-13 Assessments -- PNI homepage Accessibility (failed) (1)
Accessibility Tests -- PNI Page Assessments -- macbook-13 Assessments -- PNI homepage Accessibility (failed)

**Comment by Pomax** [comment]
Right, so: ignoring AXE entirely, this is based on Firefox's accessibility report:

image

That link goes to https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Text_labels_and_names?utm_source=devtools&utm_medium=a11y-panel-checks-text-label#Figures_with_optional_captions_should_be_labeled which suggests that we should have a figcaption inside figures: we do, but we've put it inside an `<a>` rather than the other way around, so this PR fixes that particular problem.

**Pull request merged**